src/pages/workspaces/workspace/applications/AppLauncher.js

```diff
@@ -21,14 +21,39 @@ const ApplicationLauncher = _.flow(
     breadcrumbs: props => breadcrumbs.commonPaths.workspaceDashboard(props),
     title: _.get('application')
   })
-)(({ namespace, name, refreshRuntimes, runtimes, persistentDisks, application, workspace }, ref) => {
+)(({ namespace, name, refreshRuntimes, runtimes, persistentDisks, application, workspace, workspace: { workspace: { googleProject, bucketName } } }, ref) => {
   const cookieReady = Utils.useStore(cookieReadyStore)
   const [showCreate, setShowCreate] = useState(false)
   const [busy, setBusy] = useState(false)
+  const [isWelderInit, setIsWelderInit] = useState(application !== 'RStudio') //we've already init welder if app isn't rstudio
```

Comment:
Since we also have Galaxy apps, would it be more precise to say the following?

```suggestion
  const [isWelderInit, setIsWelderInit] = useState(application !== 'RStudio') // We've already init Welder if app is Jupyter.
```